src/OutputFormatter/JUnitOutputFormatter.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace SensioLabs\Deptrac\OutputFormatter;
+
+use Fhaculty\Graph\Vertex;
+use SensioLabs\AstRunner\AstMap;
+use SensioLabs\Deptrac\ClassNameLayerResolverInterface;
+use SensioLabs\Deptrac\Dependency\Result;
+use SensioLabs\Deptrac\DependencyContext;
+use SensioLabs\Deptrac\DependencyResult\DependencyInterface;
+use SensioLabs\Deptrac\DependencyResult\InheritDependency;
+use SensioLabs\Deptrac\RulesetEngine\RulesetViolation;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Jan Schädlich <janschaedlich@sensiolabs.de>
+ */
+class JUnitOutputFormatter implements OutputFormatterInterface
+{
+    private static $argument_dump_xml = 'dump-xml';
+
+    public function getName(): string
+    {
+        return 'junit';
+    }
+
+    /**
+     * @return OutputFormatterOption[]
+     */
+    public function configureOptions(): array
+    {
+        return [
+            OutputFormatterOption::newValueOption(static::$argument_dump_xml, 'path to a dumped xml file', ''),
+        ];
+    }
+
+    public function finish(
+        DependencyContext $dependencyContext,
+        OutputInterface $output,
+        OutputFormatterInput $outputFormatterInput
+    ) {
+
+        $xml = $this->createXml($dependencyContext);
+
+        if ($dumpXmlPath = $outputFormatterInput->getOption(static::$argument_dump_xml) ?: './junit-report.xml') {
+            file_put_contents($dumpXmlPath, $xml);
+            $output->writeln('<info>JUnit Report dumped to '.realpath($dumpXmlPath).'</info>');
+        } else {
+            $output->writeln('<info>JUnit Report dump:</info>');
+            $output->writeln($xml);
+        }
+    }
+
+    private function createXml(DependencyContext $dependencyContext)
+    {
+        if (!class_exists("DomDocument")) {
+            throw new \Exception('Unable to create xml file (php-xml needs to be installed)');
+        }
+
+        $xmlDoc = new \DOMDocument('1.0', "UTF-8");
+        $xmlDoc->formatOutput = true;
+
+        $this->addTestSuites($dependencyContext, $xmlDoc);
+
+        return $xmlDoc->saveXML();
+    }
+
+    private function addTestSuites(DependencyContext $dependencyContext, \DOMDocument $xmlDoc)
+    {
+        $testSuites = $xmlDoc->createElement('testsuites');
+
+        $xmlDoc->appendChild($testSuites);
+
+        $this->addTestSuite($dependencyContext, $xmlDoc, $testSuites);
+    }
+
+    private function addTestSuite(DependencyContext $dependencyContext, \DOMDocument $xmlDoc, \DOMElement $testSuites)
+    {
+        $layers = $dependencyContext->getClassNameLayerResolver()->getLayers();
+
+        $layerIndex = 0;
+        foreach ($layers as $layer) {
+            $violationsByLayer = $dependencyContext->getViolationsByLayerName($layer);
+            if (count($violationsByLayer) === 0) {
```

Comment:
empty

Reply:
`empty` is to weak. empty could be `""`, `0`, `"0"`, `null`, `false` or `array()`.
I avoid empty, because you never know exactly what everything is checked.

👍 for `count`

Reply:
all right :)